server/portal/apps/workspace/managers/user_applications.py

```diff
@@ -286,12 +299,15 @@ def set_system_definition(
         if not system.available:
             system.enable()
 
-        settings_key = system.login.host.split('.')[0]
-        exec_settings = settings.PORTAL_EXEC_SYSTEMS.get(settings_key, {})
+        exec_settings_key = system.login.host.split('.')[0]
```

Comment:
Some people (Joe stubbs and matt vaughn in particular) will make apps that use CLI/login nodes on systems - for example, login.stampede2.tacc.utexas.edu. It may be better to do a more exact search, perhaps include a more fully qualified hostname in PORTAL_EXEC_SYSTEMS such as "frontera.tacc.utexas.edu" and search for that as a fragment of the host specified in the exec system you want to clone.

Reply:
Good point, I'll do that